src/flow/darcy_flow_lmh.cc

```diff
@@ -122,19 +121,22 @@ const it::Record & DarcyLMH::get_input_type() {
 
     DarcyLMH::EqFields eq_fields;
     
-    return it::Record("Flow_Darcy_LMH", "Lumped Mixed-Hybrid solver for saturated Darcy flow.")
+    return it::Record(equation_name(), "Lumped Mixed-Hybrid solver for saturated Darcy flow.")
 		.derive_from(DarcyFlowInterface::get_input_type())
         .copy_keys(EquationBase::record_template())
         .declare_key("gravity", it::Array(it::Double(), 3,3), it::Default("[ 0, 0, -1]"),
                 "Vector of the gravity force. Dimensionless.")
+        .declare_key("user_fields", it::Array(DarcyLMH::EqFields().make_user_field_type(equation_name())),
```

Comment:
Move key definition into EquationBase.
Create there a method for processing this key and calling init_user_fields.